content/actions/using-workflows/workflow-commands-for-github-actions.md

```diff
@@ -368,6 +368,91 @@ jobs:
 
 {% endpowershell %}
 
+### Example: Masking a generated output within a single job
+
+> **Note**
+> The order matters here, you *must* use `add-mask` before you use `set-output`.
+
+As long as you don't need to pass your secret from one job to another job, you can:
+1. Generate the secret (without outputting it).
+2. Tag it with `add-mask`.
+3. Use `set-output` to make it available to other steps within your job.
+
+{% bash %}
+
+```yaml{:copy}
+on: push
+jobs:
+  generate-a-secret-output:
+    runs-on: ubuntu-latest
+    steps:
+      - id: sets-a-secret
+        name: Generate, mask, and output a secret
+        run: |
+          the_secret=$((RANDOM))
+          echo "::add-mask::$the_secret"
+          echo "::set-output name=secret-number::$the_secret"
+      - name: Use that secret output (protected by a mask)
+        run: |
+          echo "the secret number is ${{ steps.sets-a-secret.outputs.secret-number }}"
+```
+
+{% endbash %}
+
+### Scenario: Passing a secret between jobs or workflows
+
+#### Setup
+1. Set up a secret-store to be responsible for holding secrets (e.g. Vault).
```

Comment:
I haven't tested this out, but is using the [GitHub API to create secrets](https://docs.github.com/en/rest/actions/secrets#create-or-update-a-repository-secret) not an option? I don't recall whether you can retrieve a secret that you created after the workflow run started.